File: programmsers/완전탐색/카펫.py
def solution(brown, yellow):
    total = brown + yellow  # xy

    for height in range(1, total+1):  # 1부터 total까지 돌면서
        width = total / height
        if width >= height:
            if 2*width + 2*height - 4 == brown:
                return [width, height]


brown = 10
yellow = 2
print(solution(brown, yellow))

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        //yellow의 경우 x-2 * y-2 임
        ///brown의 경우 2 * x + 2* (y-2)임
        //1500번을 돌리면=> 22500000
        for(int i = 0; i < 2000; i++) { //x가 될 친구
            for(int j = 0 ; j < 2000; j++) {
                if(i >= j) {
                    if((i-2) * (j-2) == yellow && (2 * i + 2 * (j-2)) == brown) {
                    answer[0] = i;
                    answer[1] = j;
                    break;
                }   
                }
            }
        }
        return answer;
    }
}